src/traces/treemap/style.js

```diff
@@ -36,37 +36,55 @@ function styleOne(s, pt, trace, opts) {
     var lineColor;
     var lineWidth;
     var opacity;
-
-    var depthFade = function(n) {
-        var base = trace.marker.opacitybase;
-        var step = trace.marker.opacitystep;
-
-        return n === 0 ? base :
-            Math.max(0, Math.min(1, n * step));
-    };
+    var fillColor = cdi.color;
 
     if(hovered) {
         lineColor = trace._hovered.marker.line.color;
         lineWidth = trace._hovered.marker.line.width;
         opacity = trace._hovered.marker.opacity;
     } else {
-        if(helpers.isHierarchyRoot(pt)) {
+        var isRoot = helpers.isHierarchyRoot(pt);
+
+        if(!pt.onPathbar && !trace._hasColorscale && trace.marker.depthfade) {
+            var fadedColor = Color.combine(Color.addOpacity(trace._backgroundColor, 1), fillColor);
```

Comment:
Please note that we could set the `1` on this line to 0.5 or another number if the fading effect is too much.

Reply:
These would be the baselines if we set this to `0.5`:

https://github.com/plotly/plotly.js/compare/treemap-sunburst-colors-in-colorscale...treemap-sunburst-colors-in-colorscale-etpinard

I think I prefer 0.5 over 1, but maybe something in-between would be best.

Reply:
With `0.75`:

https://github.com/plotly/plotly.js/compare/treemap-sunburst-colors-in-colorscale...treemap-sunburst-colors-in-colorscale-etpinard2

What do you think @archmoj ?

Reply:
I like `0.75` - still gives good visibility to very deep levels but a clear fade while you're near the top. 🏆

Reply:
Done in https://github.com/plotly/plotly.js/pull/4242/commits/f068cb22f75d124f5428391e230b6a4d36506c39

Thank you very much @alexcjohnson for taking a :eye: at it!